compiler/rustc_typeck/src/check/fn_ctxt/_impl.rs

```diff
@@ -879,7 +879,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             .or_else(|error| {
                 let result = match error {
                     method::MethodError::PrivateMatch(kind, def_id, _) => Ok((kind, def_id)),
-                    _ => Err(ErrorReported),
+                    _ => Err(ErrorReported::new_i_know_what_im_doing()),
```

Comment:
The `ErrorReported` should come from `MethodError` variants that actually produced errors.